src/phpDocumentor/Pipeline/Stage/Parser/ParseFiles.php

```diff
@@ -41,19 +33,17 @@ public function __construct(
         $this->reEncodingMiddleware = $reEncodingMiddleware;
     }
 
-    public function __invoke(Payload $payload): Payload
+    public function __invoke(ApiSetPayload $payload): ApiSetPayload
     {
-        /*
-         * For now settings of the first api are used.
-         * We need to change this later, when we accept more different things
-         */
-        $apiConfig = current($payload->getApiConfigs());
-        if ($apiConfig === false) {
-            return $payload;
-        }
+        $apiConfig = $payload->getApiSet()->getSettings();
 
         $builder = $payload->getBuilder();
-        $builder->setApiSpecification($apiConfig);
+        $builder->usingApiSpecification($apiConfig);
+        $builder->usingDefaultPackageName($apiConfig['default-package-name'] ?? '');
+
+        // TODO: The setVisibility call should purge the cache if it differs; but once we are here, cache has already
```

Comment:
I think when we move the filtering into the compiler... we do not need to invalidate the cache... filtering should just transform the output of the parser stage, it will not harm to have a pure parser output. As the filters are already running over the descriptors.

Reply:
IIRC, the visibility part was originally in the parser instead of the compiler so that 'unauthorized' code would not end up in the cache or in the ast dump file. This as a means of protecting IP even though things could be distributed. I am doubting whether it is a real thing, but that was the original reason :)